src/components/Person/Person.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const Person = ({ name, age, partherName, sex, isMarried }) => {
+  let pathner;
+  let personPartner;
+  let personAge;
+
+  if (age === undefined) {
+    personAge = 'Age is hidden';
+  } else {
+    personAge = `I am a ${age}`;
+  }
+
+  if (isMarried === true && sex === 'm') {
+    pathner = 'wife';
+  } else if (isMarried === true && sex === 'f') {
+    pathner = 'husband';
+  }
+
+  if (isMarried === true) {
+    personPartner = `My ${pathner}'s name is ${partherName}`;
+  } else if (isMarried === false) {
+    personPartner = 'I am not married';
+  }
```

Comment:
It's better to use ternary operator if you want set some value to variable depending on some check